cmake/Date.cmake

```diff
@@ -69,6 +69,9 @@ if (NOT date_src_POPULATED)
     add_library(date::tz ALIAS date-tz)
     target_include_directories(date-tz PUBLIC ${DATE_INCLUDE_DIR})
     target_compile_features(date-tz PUBLIC cxx_std_11)
+    if (NOT WIN32)
+        target_compile_options(date-tz PRIVATE "-Wno-error")
```

Comment:
I tried that at first, but it seemed to be way more problematic as a lot of CMakeFiles needed to be updated for that. The other reason I decided to do it this way is because code added in the future will be automatically have the warning compile options, without the need to explicitly define them again. In the third party libraries they are not part of the PASSTHROUGH cmake options, only the Date.cmake file had to be updated with this option as they do not use those options.